e2e/LongHaul/service/IotHub.cs

```diff
@@ -68,125 +68,129 @@ public async Task MonitorConnectedDevicesAsync(CancellationToken ct)
         {
             while (!ct.IsCancellationRequested)
             {
-                AsyncPageable<ClientTwin> allDevices = s_serviceClient.Query.Create<ClientTwin>(
-                    "SELECT deviceId, connectionState FROM devices where is_defined(properties.reported.runId)",
-                    ct);
-
-                AsyncPageable<ClientTwin> allModules = s_serviceClient.Query.Create<ClientTwin>(
-                    "SELECT deviceId, moduleId, connectionState FROM devices.modules where is_defined(properties.reported.runId)",
-                    ct);
-
-                await foreach (ClientTwin device in allDevices)
+                try
```

Comment:
Add a try/catch as I found this failed if I killed the connection and ended the task.